applications/mne_sample_set_downloader/downloader.h

```diff
@@ -0,0 +1,141 @@
+*       to endorse or promote products derived from this software without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+* WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+* PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+* INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+* PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+* HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+* NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+*
+* @brief     Downloader class declaration.
+*
+*/
+
+#ifndef DOWNLOADER_H
+#define DOWNLOADER_H
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// INCLUDES
+//=============================================================================================================
+
+#include "Extract.h"
+
+//*************************************************************************************************************
+//=============================================================================================================
+// QT INCLUDES
+//=============================================================================================================
+
+#include <QObject>
+#include <QMainWindow>
+#include <QNetworkAccessManager>
+#include <QNetworkReply>
+#include <QUrl>
+#include <QString>
+#include <QtCore>
+#include <QFile>
+#include <QSharedPointer>
+#include <QDebug>
+#include <QIODevice>
+#include <QFileDialog>
+
+//*************************************************************************************************************
+//=============================================================================================================
+// DEFINE NAMESPACE DOWNLOADER
+//=============================================================================================================
+
+using namespace std;
+
+namespace Ui {
+class Downloader;
+}
+
+/**
+* This class builds a Downloader Object an GUI. It downloads the sample data set and stores it in a temporary file
+* for the Extract class to extract once it recieves the download_finished signal.
+* The temporary file is deleted on leaving the GUI. The class also handles error and finalisation signals
+* from the extract class and adjusts the GUI accordingly (i.e. displays "Done" on finalization and ask if/where 7zip is
+* installed if the Extract class can't find it).
+*
+* @brief Downloads the sample data set and manages the GUI
```

Comment:
This is the description of the overall file. Should say: Downloader class declaration.